lib/logstash/outputs/pravega.rb

```diff
@@ -26,71 +26,63 @@ class LogStash::Outputs::Pravega < LogStash::Outputs::Base
 
   config :password, :validate => :string, :default => ""
 
+
   public
   def register
+     @clientFactory = get_client_factory
+     @producer = create_producer
   end # def register
 
   public
   def multi_receive_encoded(encoded)
-    pre_check(encoded)
-    @producer = create_producer
     encoded.each do |event,data|
       begin
         @producer.writeEvent(routing_key,data)
-	logger.debug("write in stream succssfully", :stream_name => @stream_name, :event => data)
+        logger.debug("write in stream succssfully", :stream_name => @stream_name, :event => data)
       rescue LogStash::ShutdownSignal
         logger.debug("Pravega producer got shutdown signal")
       rescue => e
         logger.warn("Pravega producer threw exception, restarting", :exception => e)
       end
     end
-    @producer.close()
   end
 
-  private
-  def pre_check(encoded)
-    # If the num_of_segment <= 0, the IllegalArgumentException will be thrown and logstash will stop.
-    @num_of_segments = 1 if @num_of_segments <= 0
-
-    # If the input stream name is same as output, it will filter the formatted json data again and again without endless.
-    # So, the new output streamName needs to be created
-    encoded.each do |event, data|
-      if @stream_name == JSON.parse(data)['streamName']
-        @stream_name = "newOutputStream-".concat(SecureRandom.uuid)
-	logger.info("The input stream_name shouldn't be equal to output, the new output_stream is created", :stream_name => @stream_name)
-	break
-      end
-    end
-    logger.debug("After arugument preCheck ", :num_of_segments => @num_of_segments, :stream_name => @stream_name)
+  def close
+    @clientFactory.close()
```

Comment:
clientFactory.close() should come after producer.close(). Close order should be opposite of construction order.